includes/model/class-user.php

```diff
@@ -281,9 +281,9 @@ public function get_attachment() {
 					$tags->next_tag();
 				}
 
-				if ( 'A' === $tags->get_tag() ) {
+				if ( 'A' === $tags->get_tag() && $tags->get_attribute( 'href' ) ) {
 					$tags->set_bookmark( 'link' );
-					if ( ! $tags->next_tag() ) {
+					if ( ! $tags->next_tag() || 'SPAN' === $tags->get_tag() ) {
```

Comment:
why the `SPAN` check here?

Reply:
When youn not check for `SPAN` it will ignore tags like `<a href="test.de><span>Tiltle</span></a>"`. So it will also work when you generate links like mastodon.

Reply:
Is `$tags->next_tag()` the next tag on the same level or one level down the DOM?

I thought it is on the same level as the Link!

Reply:
On my tests `$tags->next_tag()` is simply the next tag after the opening 'a' tag. 

When you use  `!$tags->next_tag('A')` you can be shure that there are not more `a` tags in the DOM. 
Depends on what the idea was behind :
Checking ther is only one `a` tag and getting the link from it, or also checking the the Tag has only text.

Reply:
OK, then my initial code is wrong! My idea was to check if there is only one link in the text!

Reply:
OK, i can fix that in the PR. So that it is like that.

Reply:
thanks @danielhuesken !